api/v1alpha1/applicationset_types.go

```diff
@@ -21,6 +21,12 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Utility struct for a reference to a secret key.
+type SecretRef struct {
+	Name string `json:"name"`
```

Comment:
`name` is how this is handled in corev1 so I would probably leave it there? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#secretkeyselector-v1-core

It looks like:

```yaml
secretRef:
  name: foo
  key: bar
```

Reply:
Though as I'm thinking of other places that do this, secret volumes use `secretName` so no actual consistency either way. Whatever you prefer.

Reply:
Agreed, definitely no consistency, I went looking to see if there was any pre-existing idiom here that we could draw upon and likewise came up short. 

Let's go with `secretName` here, I think this makes it more obvious where that value is coming from (which is otherwise not indicated/implied via context).

Reply:
Done